src/node_api.cc

```diff
@@ -469,6 +477,9 @@ v8::Local<v8::Object> CreateFunctionCallbackData(napi_env env,
       v8impl::kFunctionIndex,
       v8::External::New(isolate, reinterpret_cast<void*>(cb)));
   cbdata->SetInternalField(
+      v8impl::kEnvIndex,
+      v8::External::New(isolate, static_cast<void*>(env)));
```

Comment:
Move this up to before setting the function, just so things are in a consistent order.